components/fxa-client/ffi/src/lib.rs

```diff
@@ -546,7 +561,7 @@ pub unsafe extern "C" fn fxa_initialize_device(
     log::debug!("fxa_initialize_device");
     ACCOUNTS.call_with_result_mut(error, handle, |fxa| {
         let capabilities =
-            DeviceCapability::from_protobuf_array_ptr(capabilities_data, capabilities_len);
+            DeviceCapability::from_protobuf_array_ptr(capabilities_data, capabilities_len)?;
```

Comment:
This now errors out instead of panic'ing (Was this an okay change?)

Reply:
After Slack discussion, it should be fine.